web-admin/src/features/projects/status/waitForDeployment.ts

```diff
@@ -0,0 +1,221 @@
+      });
+      void goto(`/${this.organization}/${this.project}`);
+    } else if (this.redirectToResource) {
+      let dashboardLink = "";
+      switch (this.redirectToResource.meta?.name?.kind) {
+        case ResourceKind.MetricsView:
+          dashboardLink = `/${this.organization}/${this.project}/${this.redirectToResource.meta.name.name}`;
+          break;
+
+        case ResourceKind.Dashboard:
+          dashboardLink = `/${this.organization}/${this.project}/-/dashboards/${this.redirectToResource.meta.name.name}`;
+          break;
+
+        // any new visual resource would need to be added here
+      }
+
+      eventBus.emit("notification", {
+        message: "Project shouldRedirect successfully",
+        ...(dashboardLink
+          ? {
+              link: {
+                text: "Go to dashboard",
+                href: dashboardLink,
+              },
+            }
+          : {}),
+      });
+    }
+  }
+}
+
+function deploymentListener(
+  organization: string,
+  project: string,
+): CompoundQueryResult<string> {
+  return derived(
+    useRefetchingProject(organization, project),
+    (projectResp, set) => {
+      if (projectResp.isFetching) {
+        set({
+          isFetching: true,
+          error: undefined,
+        });
+        return;
+      }
+
+      if (
+        projectResp.data?.prodDeployment?.status ===
+        V1DeploymentStatus.DEPLOYMENT_STATUS_ERROR
+      ) {
+        set({
+          isFetching: false,
+          error:
+            projectResp.data.prodDeployment.statusMessage ??
+            "deployment failed",
+        });
+        return;
+      }
+
+      const instanceId =
+        projectResp.data?.prodDeployment?.runtimeInstanceId ?? "";
+
+      derived(useRefetchingProjectParser(instanceId), (projectParserResp) => {
+        if (
+          projectParserResp.isLoading ||
+          projectParserResp.data?.meta?.reconcileStatus !==
+            V1ReconcileStatus.RECONCILE_STATUS_IDLE
+        ) {
+          return {
+            isFetching: true,
+            error: undefined,
+          };
+        }
+
+        return {
+          isFetching: false,
+          error: projectParserResp.data?.meta?.reconcileError,
+          data: instanceId,
+        };
+      }).subscribe(set);
+    },
+  );
+}
+
+function useRefetchingProject(organization: string, project: string) {
+  return createAdminServiceGetProject<V1GetProjectResponse>(
+    organization,
+    project,
+    undefined,
+    {
+      query: {
+        refetchInterval: (data) => {
+          if (
+            !data?.prodDeployment?.status ||
+            data.prodDeployment.status ===
+              V1DeploymentStatus.DEPLOYMENT_STATUS_UNSPECIFIED ||
+            data.prodDeployment.status ===
+              V1DeploymentStatus.DEPLOYMENT_STATUS_PENDING
+          ) {
+            return PollTime;
+          }
+          return false;
+        },
+        queryClient,
+      },
+    },
+  );
+}
```

Comment:
Looks like this might be redundant to the `GetProject` call in `[org]/[project]/+layout.svelte`?